test/test_unit_kernel.cpp

```diff
@@ -9,17 +9,49 @@
 #include "gtest/gtest.h"
 
 #include <string>
+#include <memory>
 
 #include "nlohmann/json.hpp"
 
+#include "test_interpreter.hpp"
+
 #include "xeus/xkernel.hpp"
 #include "xeus/xmiddleware.hpp"
 #include "xeus/xsystem.hpp"
 
+#include "xeus/xhelper.hpp"
+
 namespace nl = nlohmann;
 
 namespace xeus
 {
+    // TEST(kernel, print_starting_message)
+    // {
+    //     std::string file_name = "connection.json";
+    //     xconfiguration config = load_configuration(file_name);
+
+    //     using interpreter_ptr = std::unique_ptr<test_kernel::test_interpreter>;
+    //     interpreter_ptr interpreter = interpreter_ptr(new test_kernel::test_interpreter());
+    //     xkernel kernel(config, get_user_name(), std::move(interpreter));
+
+    //     print_starting_message(config);
+    // }
```

Comment:
I don't understand how to initialize a Kernel. Maybe the problem here is because I'm using this `connection.json` that doesn't exist? I thought it should work regardless because it has default values. Anyway, I'm getting the following when trying to compile:

```
(xeus)  ✘  ~/dev/xeus/build   output ●✚  make install -j12
[  2%] Built target gtest
[  4%] Built target gtest_main
[ 81%] Built target xeus
[ 84%] Built target xeus-static
Consolidate compiler generated dependencies of target test_unit_kernel
[ 86%] Built target test_xin_memory_history_manager
[ 90%] Built target test_kernel_shell
[ 93%] Built target test_kernel
[ 97%] Built target test_kernel_control
[ 98%] Building CXX object test/CMakeFiles/test_unit_kernel.dir/test_unit_kernel.cpp.o
/Users/mariana/dev/xeus/test/test_unit_kernel.cpp:51:19: warning: ISO C++11 does not allow conversion from string literal to 'char *' [-Wwritable-strings]
        argv[0] = "--version";
                  ^
1 warning generated.
[100%] Linking CXX executable test_unit_kernel
Undefined symbols for architecture x86_64:
  "vtable for test_kernel::test_interpreter", referenced from:
      xeus::kernel_print_starting_message_Test::TestBody() in test_unit_kernel.cpp.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [test/test_unit_kernel] Error 1
make[1]: *** [test/CMakeFiles/test_unit_kernel.dir/all] Error 2
make: *** [all] Error 2
```

Can you help me with this @JohanMabille? :)